data/projects/documentation_extraction.md

```diff
@@ -0,0 +1,33 @@
+Code needs documentation. Most programming languages have some form of inline 
+documentation standard.
+
+For python, its [docstrings](https://www.python.org/dev/peps/pep-0257/)
+
+The aim of the project is to develop a language independent analyzing routine 
+that will find all documentation strings in a file, parse them into specific 
+groups and check them against a specified documentation style given by the 
+user. The functionality can be extended so that the analyzing routine provides
+a patch that re-formats the documentation correctly, check for grammatical 
+errors and perform other aesthetical fixes.
+
+
+#### Milestones
+
+##### GSOC 2017 COMMUNITY BONDING
+
+* Learn about how different languages have their own documentation standards.
+* Learn about the work already done.
+* Design an effecient parsing system.
+
+##### GSOC 2017 MIDTERM
+
+* Make DocumentationStyleBear work for atleast one language, preferably python.
+* Create a base class that does the extraction and parsing for us. 
```

Comment:
Milestones should describe a state that is/has to be reached.
So this could be:!A base class, that does the extraction and parsing, is implemented. Fixing <issue link>
and the one below could be:"<issue link> is fixed. with context what it is about.